pkg/types/openstack/machinepool.go

```diff
@@ -49,6 +54,13 @@ func (o *MachinePool) Set(required *MachinePool) {
 	if required.AdditionalSecurityGroupIDs != nil {
 		o.AdditionalSecurityGroupIDs = append(required.AdditionalSecurityGroupIDs[:0:0], required.AdditionalSecurityGroupIDs...)
 	}
+
+	if len(required.Zones) > 0 {
+		o.Zones = required.Zones
+	} else {
+		o.Zones = []string{""}
+	}
```

Comment:
Why is this being set to array with one empty element?? I don't think this is good. Handle empty zone list as a valid input.

Reply:
handling it here is esp not correct.